File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/behavior/BaseBottomSheetCallback.java
package ru.testsimpleapps.coloraudioplayer.ui.behavior;

import android.support.annotation.NonNull;
import android.support.design.widget.BottomSheetBehavior;
import android.util.Log;
import android.view.View;

public abstract class BaseBottomSheetCallback extends BottomSheetBehavior.BottomSheetCallback {

    public static final String TAG = BaseBottomSheetCallback.class.getSimpleName();
    protected float mDelta;
    protected int mState;

    @Override
    public void onStateChanged(@NonNull View view, int i) {
        Log.d(TAG, "onStateChanged() - state: " + i);
        mState = i;
    }

    @Override
    public void onSlide(@NonNull View view, float v) {
        Log.d(TAG, "onSlide() - delta: " + v);
        mDelta = v;
    }

}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/behavior/BaseBottomSheetCallback.kt
package ru.testsimpleapps.coloraudioplayer.ui.behavior

import android.support.design.widget.BottomSheetBehavior
import android.util.Log
import android.view.View

abstract class BaseBottomSheetCallback : BottomSheetBehavior.BottomSheetCallback() {
    protected var mDelta: Float = 0.toFloat()
    protected var mState: Int = 0

    override fun onStateChanged(view: View, i: Int) {
        Log.d(TAG, "onStateChanged() - state: $i")
        mState = i
    }

    override fun onSlide(view: View, v: Float) {
        Log.d(TAG, "onSlide() - delta: $v")
        mDelta = v
    }

    companion object {

        val TAG = BaseBottomSheetCallback::class.java.simpleName
    }

}
